Automattic/wp-calypso
# Site Editor: maintain calypso origin through testing envs

Related to #75183 

## Proposed Changes

Helps manage consistency between Calypso and the Site Editor by ensuring that returning nav links to the same Calypso environment. This will allow us to test #75183 outside of only `https://wordpress.com`.

## Testing Instructions

1. sandbox `widgets.wp.com` and whichever site you will be testing against
2. Apply D107058-code or run `install-plugin.sh wbe update/maintain-calypso-origin-in-site-editor` on your wpcom sandbox
3. Load the Site Editor **outside** of the iframe (Ideas: use #75183 locally to be redirected without thought, otherwise use `/wp-admin/site-editor.php` on the sandboxed site or even manually grab the URL from the iframe in devtools)
4. You should be linked back to the same testing environment from the nav sidebar's back button: 

<img width="358" alt="Screenshot 2023-03-28 at 16 29 09" src="https://user-images.githubusercontent.com/195089/228371212-82381ebb-0615-4a30-98a3-1ef01908751b.png">


## Pre-merge Checklist
